Physicc/include/bvh.hpp

```diff
@@ -24,17 +24,25 @@ namespace Physicc
 			inline void buildTree();
 			//build a tree of the bounding volumes
 
-			std::vector<RigidBody>& convert();
 			//convert the tree into a linear data structure
+			std::vector<RigidBody>& convert();
 
 		private:
 			BVHNode* m_head;
 			std::vector<RigidBody> m_rigidBodyList;
 
-			using Iterator = std::vector<RigidBody>::iterator;
+			BoundingVolume::AABB computeBV(int start, int end);
+
+			void buildTree(BVHNode* node, int start, int end);
+
+			enum Axis {
+				X,
+				Y,
+				Z,
+			};
 
-			BoundingVolume::AABB computeBV(Iterator begin, Iterator end);
-			void buildTree(BVHNode* node, Iterator begin, Iterator end);
+			void sort(Axis axis, int start, int end);
```

Comment:
Same for these two functions: I want them to use `std::size_t` instead of `int`.